### PR TITLE
Add capability to skip GH status check

### PR DESCRIPTION
**Pull request by tspascoal**
If environment variable GEI_SKIP_STATUS_CHECK is set true, no GH status check will be performed

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)
- [x] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/ado2gh/Program.cs
Comment (ArinGhazarian) [inline src/ado2gh/Program.cs:99]: Two suggestions:
1. Would you put the env variables in [EnvironmentVariableProvider](https://github.com/github/gh-gei/blob/bb3c45aa895eb5ce67d94de115288ccee3a22908/src/Octoshift/Services/EnvironmentVariableProvider.cs. It'd be easier to manage all env variables the cli uses. In this case there is no need to register it as a secret of course. 
2. It would probably be a good idea to also support "1" as a valid valid value so the check would be:
```c#
Environment.GetEnvironmentVariable("GEI_SKIP_STATUS_CHECK")?.ToUpperInvariant() is "TRUE" or "1"
```

Comment (tspascoal) [reply src/ado2gh/Program.cs:99]: I did consider both of these options, but ended up following the same pattern as this [prior art](https://github.com/github/gh-gei/blob/ddefd7ac385f38fa4f37f57183a19d40bebe1c62/src/Octoshift/Services/OctoLogger.cs#L40) for consistency

Either way works for me tell me which one makes more sense

Comment (ArinGhazarian) [reply src/ado2gh/Program.cs:99]: I still think putting everything is the `EnvironmentVariableProvider` makes more sense since it acts like a central place for reading env. variables. Regarding whether to support "1", I'll leave it to you it doesn't make that much difference anyways.

Comment (tspascoal) [reply src/ado2gh/Program.cs]: Change the code in 48db53cec7434ca6b73426a6cf767814d4cbc096 to use the `EnvironmentVariableProvider`

Added a `GetValue` to get the value and not mask the value, `GetSecret` uses this method as well

Once this is merged will refactor #1154 to use the same pattern as well

File: src/Octoshift/Services/EnvironmentVariableProvider.cs
Comment (ArinGhazarian) [inline src/Octoshift/Services/EnvironmentVariableProvider.cs:78]: nit: newline

File: src/Octoshift/Services/EnvironmentVariableProvider.cs
Comment (ArinGhazarian) [inline src/Octoshift/Services/EnvironmentVariableProvider.cs:68]: nit: `value.IsNullOrWhiteSpace()`

File: src/Octoshift/Services/EnvironmentVariableProvider.cs
Comment (ArinGhazarian) [inline src/Octoshift/Services/EnvironmentVariableProvider.cs:67]: I ran `dotnet format` and I also didn't like the (so called) simplified version.

**Review by ArinGhazarian — COMMENTED** [review]
Looks good, just added a couple of nits.

File: src/Octoshift/Services/EnvironmentVariableProvider.cs
Comment (tspascoal) [reply src/Octoshift/Services/EnvironmentVariableProvider.cs:78]: Addressed in ef5595a23276fcc052a063b906efeb0c1efe2500

File: src/Octoshift/Services/EnvironmentVariableProvider.cs
Comment (tspascoal) [reply src/Octoshift/Services/EnvironmentVariableProvider.cs]: Addressed this in 13ba4f10a2a2318beeb4acc7519ffdf9e5ff5441

Since this changes the env variable semantic slightly also added a unit test

**Review by ArinGhazarian — APPROVED** [review]

File: README.md
Comment (timrogers) [inline README.md:92]:
```suggestion
When the CLI is launched, it logs a warning if there are any ongoing [GitHub incidents](https://www.githubstatus.com/) that might affect your use of the CLI. You can skip this check setting the `GEI_SKIP_STATUS_CHECK` environment variable to `true`. 
```

File: src/ado2gh/Program.cs
Comment (timrogers) [inline src/ado2gh/Program.cs:103]:
```suggestion
                Logger.LogInformation("Skipped GitHub status check due to GEI_SKIP_STATUS_CHECK environment variable");
```

File: src/bbs2gh/Program.cs
Comment (timrogers) [inline src/bbs2gh/Program.cs:110]:
```suggestion
                Logger.LogInformation("Skipped GitHub status check due to GEI_SKIP_STATUS_CHECK environment variable");
```

File: src/gei/Program.cs
Comment (timrogers) [inline src/gei/Program.cs:103]:
```suggestion
                Logger.LogInformation("Skipped GitHub status check due to GEI_SKIP_STATUS_CHECK environment variable");
```

**Review by timrogers — APPROVED** [review]
This is looking great ✨ I'm going to make a couple of tweaks to the readme and log output, and then merge.

**Pull request merged**
